Add team icons

SashaBajzek opened the pull request:
This is a work in progress PR for Trello Card:  https://trello.com/c/jjEeir6V/107-add-social-media-icons-mail-button-under-each-persons-face
I added the linkedin and email social icons reusing the react-social-icons library that you had used for the purple social icons at the top of the page.
I do not see functionality for an onhover / onfocus state for the icon colors to be inverted, so we might need to bring in our own icons if you want that.
Also, I did not map through the array of social links as we had discussed in the standup since the email links require the 'mailto' to be added to the URL.
I deployed to dev-1 so you could see how the icons look.  I added Snorre's email in the Sanity dashboard so you could have at least one example with the email icon.  I don't know anyone else's email to add to the CMS.

vongohren commented:
Looks good and simple even if we didn't do a component.

SashaBajzek commented:
I updated the color and size per Marina's feedback.  Deployed to: https://diwala-frontpage-dev.firebaseapp.com/

vongohren commented on src/components/team/Team.tsx:
```diff
@@ -13,6 +14,8 @@ export default function Team(props: any) {
 
   // const imageWidth = Math.round(getViewportWidth() / 4);
+
+  const socialIconColor = '#580394';
```
This is the color in the sass file? Could you try to utalize what they explain in this article
https://blog.bluematador.com/posts/how-to-share-variables-between-js-and-sass/

SashaBajzek replied:
I tried to do this, but I get an error:  Property 'diwalaIconPurple' does not exist on type '(props: any) => Element'

Were you able to pull the color in JS somewhere else that I can see as an example?

vongohren commented on src/components/team/Team.tsx:
```diff
@@ -15,7 +16,8 @@ export default function Team(props: any) {
   // const imageWidth = Math.round(getViewportWidth() / 4);
 
-  const socialIconColor = '#580394';
+  // const socialIconColor = '#580394';
```
Did it work, if so we can delete this line? And just bring along the comments above as well, they dont make sense anymore

SashaBajzek replied:
It didn't work for me.  The article said that CSS modules were what made the import to JS possible.  Are CSS modules turned on for this project?  I couldn't find a config file.

SashaBajzek replied:
Looking at the footer icons and the hero icons, they were all typed into the JS file, not brought in through sass.  I agree, it would be cool to get this to work so they could all reference the same variable!

vongohren replied:
Yeah so I expect you to make sure it works before it is pushed into the repo. That you have to test locally before you add it to the PR, because i could have merged this and the it would not work :( So its important to have an insight to what is pushed to the PR :)

SashaBajzek commented:
@vongohren I followed the instructions and added the changes to webpack, but it still doesn't pull in the color variables.  Do you have an example in the code of it working with this setup that I can use as an example?

alodahl commented:
Hey @SashaBajzek, I sent you a color variable example in slack.

vongohren commented:
I dont have a concrete example because I have not done it before, it just made sense to me that it had to be doable. But the reason for all these not working is basically that we are using node-sass-chokidar command in package json to compile the scss. And not going via webpack. Googeling a bit around that lib now

vongohren commented:
This line works: 

const styles  = require('../../compiled_css/components/team/Team.css');

We have a not that great setup for our css!
Feks it should have been put in each folder.
It does not run via webpack config.
Only when we build, ref this line i package.json - "build": "npm-run-all build-css build-js build-polyfill",

We only use build polyfill once meaning that the config you were editing didnt give much affect.

But i think you can remove the current config stuff, and just use the top line.
What you can do is to create a task in the backlog section explaining some issues you have experienced. Such as import doesnt work, because it is never made a module, which is type scripts fault becuase you need to type file before they can import. That is why require is used.

We cant require the scss file itself, because the sass-loader has not been setup or dont work on node-sass-chokidar.

vongohren commented:
I keep getting this error though, have you experienced it?

<img width="644" alt="screen shot 2018-07-30 at 18 48 11" src="https://user-images.githubusercontent.com/1012055/43411005-283cc614-9429-11e8-9763-112b81240739.png">

SashaBajzek commented:
@vongohren thanks for the clarification on the webpack / node-sass-chokidar setup!  I can see now how there was no structure to pull straight from the scss file.
I switched out the scss require for the css file and now it pulls in the color variable just fine.  I also removed the extra <a> tag that was causing the warning you were seeing.  I deployed to dev 1:  https://diwala-frontpage-dev.firebaseapp.com/

SashaBajzek commented:
I also added a card to the Trello backlog describing this:  https://trello.com/c/sknmVo7F/123-setup-clearer-scss-to-css-pipeline

vongohren commented:
Great 😄
